Nominate Danny Cao as Fabric maintainer

denyeart opened the pull request:
I would like to nominate Danny Cao as a Fabric maintainer.
Danny dove into the project last year and made significant
contributions in the areas of private data and gossip
dissemination. More recently, Danny has been leading a team
of contributors to add a library for channel configuration,
which has been a long-standing gap in the Fabric project.
Danny spends many hours coaching other contributors on
Go and Fabric internals, and has been an active PR
reviewer already.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

ratnakar-asara commented:
Signed-off-by: rasara@us.ibm.com

srderson commented:
@muralisrini @ratnakar-asara First, thanks for this PR. This is a much needed change. I notice there are changes in openchain.proto in the transaction type such as `CHAINCODE_QUERY = 4;` to `CHAINCODE_QUERY = 3;`. This would normally break compatibility. Is this a problem or is it ok as long as you're also using an updated client?

@stanliberman This PR contains the Init function in a chaincode now so it's likely of interest to you.

I know one of Stan's requirements was a guarantee that the Init function is only run one. Is that the case here?

muralisrini commented:
@srderson : chaincode messages are just between the peer and chaincode (does not bleed into user apis)... so should be fine.

Currently we allow deploy to be called for the same chaincode any number of times (recorded in #581). When that gets fixed, init of a chaincode cannot be called multiple times.

muralisrini commented:
@srderson : as part of this work @ratnakar-asara had to make APIs such as the REST APIs compatible as well. So yes, as long as clients side are uptodate we should be fine.

@angrbrd : can you please review to make sure please ?

As far calling init only once, there's issue #581 to prevent duplicate deploys. Once that's fixed, the init function can only be run once (at the first deploy).

muralisrini commented:
@ratnakar-asara : we will have to do this work on the Hyperledger codebase anyway. So please keep you branch around for us to continue there. Will close this PR on this repo so we can open a new one on Hyperledger.

muralisrini commented:
@stanliberman : we will continue with this PR under hyperledger github

denyeart committed:
Nominate Danny Cao as maintainer

I would like to nominate Danny Cao as a Fabric maintainer.
Danny dove into the project last year and made signicant
contributions in the areas of private data and gossip
dissemination. More recently, Danny has been leading a team
of contributors to add a library for channel configuration,
which has been a long-standing gap in the Fabric project.
Danny spends many hours coaching other contributors on
Go and Fabric internals, and has been an active PR
reviewer already.

Signed-off-by: David Enyeart <enyeart@us.ibm.com>

christo4ferris reviewed (approved):
LGTM!

denyeart commented:
Majority approval...merging...
